src/engraving/dom/utils.cpp

```diff
@@ -206,16 +206,19 @@ Segment* Score::tick2segment(const Fraction& tick, bool first) const
 /// the first segment *before* this tick position
 //---------------------------------------------------------
 
-Segment* Score::tick2leftSegment(const Fraction& tick, bool useMMrest, bool anySegmentType) const
+Segment* Score::tick2leftSegment(const Fraction& tick, bool useMMrest, SegmentType segmentType) const
 {
     Measure* m = useMMrest ? tick2measureMM(tick) : tick2measure(tick);
     if (m == 0) {
         LOGD("tick2leftSegment(): not found tick %d", tick.ticks());
         return 0;
     }
 
+    if (m->isMMRest()) {
+        segmentType = SegmentType::SpannerAnchor;
```

Comment:
Modifying a parameter of a method inside the body of the method seems not a very clean approach, and might make the method unpredictable: you specify a desired segment type but then the method may still end up returning a different type.
It is also a bit mysterious _why_ this is done; a comment with explanation would be good here. But I think it's better to try if a small refactor can be done so that it is no longer necessary.